src/components/Scaler/index.js

```diff
@@ -0,0 +1,108 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import * as d3 from 'd3';
+import DataContext from '../../context/Data';
+import ScalerContext from '../../context/Scaler';
+import { createXScale, createYScale } from '../../utils/scale-helpers';
+
+class Scaler extends Component {
+  static propTypes = {
+    children: PropTypes.node.isRequired,
+    dataContext: PropTypes.shape({
+      baseDomain: PropTypes.arrayOf(PropTypes.number).isRequired,
+      subDomain: PropTypes.arrayOf(PropTypes.number).isRequired,
+      subDomainChanged: PropTypes.func.isRequired,
+      series: PropTypes.arrayOf(
+        PropTypes.shape({
```

Comment:
Reuse this shape from another series descriptor

Reply:
Yep.